keras/src/models/functional.py

```diff
@@ -102,6 +102,8 @@ def __new__(cls, *args, **kwargs):
     @tracking.no_automatic_dependency_tracking
     def __init__(self, inputs, outputs, name=None, **kwargs):
         if isinstance(inputs, dict):
+            self._input_names = list(inputs.keys())
```

Comment:
IIUC is this PR relying on the order of dict keys to be deterministic in order to work? I believe this is guaranteed now in [Python >3.7](https://docs.python.org/3.7/library/stdtypes.html#typesmapping:~:text=Dictionaries%20preserve%20insertion%20order.), but maybe worth clarifying this assumption in a comment.